### ISSUE TITLE
Certain formats can't be decoded with r1050-247-g8365af7

### ISSUE THREAD

**Issue by ghf50** [kode54/vgmstream#64]
Hi, dear developers!
I can't play these formats below after I updated my vgmstream plugin to r1050-247-g8365af7, foobar2000 just keep saying:

> Decoding failure at 0:00.000 (Unsupported format or corrupted file):

- NDS strm
- GC adp
- GC dsp
- GC agsc
- Wii brstm
- Wii fsb
- Wii rwav
- PSP oma
- PSP at3
- X360 xma

**Comment by kode54** [comment]
This should be fixed by b5e6808, which will be built shortly.

**Comment by ghf50** [comment]
Thanks for you fellows' great work! most of them fixed by r1050-248-gb5e6808, 
but 

- NDS strm
- PSP at3
- PSP oma
- X360 xma

still broken.

**Comment by kode54** [comment]
That update actually broke more than it fixed, and I need to debug a proper fix to the interleaved files handler for seeking, since it seems to be doing funny things it shouldn't. But I haven't got my debug and development environment up and running, because I need to move Visual Studio into a virtual machine, since there are debugging issues with Windows 10 and GUI debugee apps that can cause the entire login session to become unresponsive to anything but ctrl-alt-del -> log out, except for Remote Desktop sessions. I'll get around to fixing this soonish.

**Comment by kode54** [comment]
This should be fixed now.

**Comment by ghf50** [comment]
Great thanks to you, kode54. You're efficiency.

**Comment by rmenessec** [comment]
I've just started using this plugin to play some STRM (Nintendo DS) files, and... it's failing on every file in a particular soundtrack. foobar2000 locks up hard and never recovers. The reported component version is r1050-481-g121226c.

What other information can I provide to help?

Thanks!

**Comment by kode54** [comment]
It would help to know exactly which soundtrack fails, and some sample file(s) to test. I can't just guess.

**Comment by rmenessec** [comment]
Right, sorry. [Advance Wars: Days of Ruin](https://www.zophar.net/music/nintendo-ds-2sf/advance-wars-days-of-ruin).

**Comment by kode54** [comment]
Are you sure it's my component? I just verified that all of the STRM and MINI2SF files both work with their respective components.

**Comment by rmenessec** [comment]
Well, first, is it possible for two components to fight over the same format in fb2k? I tried to verify the actual format handler before commenting, but I've come up with nothing. The files are unpacked; not left in the archive. That should rule out fb2k's archive input handler.

Most of the post-install modules I'm using were, well, written by you. 😉 (_Seriously_ impressive, by the way. Also, thanks!)

I removed the WASAPI output module entirely after I had a lockup while playing some very simple CBR LAME-encoded MP3s; nothing unusual. Unfortunately, that didn't solve the problem with the STRM files, and that was a few days back, before I started experimenting with DS native audio files.

I can't verify for certain that my audio device supports 16.4KHz, but I hope so. It's a real, slotted expansion card: an ASUS Xonar DX. ASUS' documentation on an obscure question like that is nonexistent. I made a quick attempt at finding a datasheet for the CMI8788 supposedly powering it and turned up [this PDF](http://www.hardwaresecrets.com/datasheets/OxygenHD%20CMI8788%20Datasheet%20Rev0.6.pdf) labeled 'v0.6' (not confidence-inspiring).

It more or less repeats what ASUS says: that it's capable of 192KHz @ 24b. It does list pinouts, I²C bus rates, and other such details; perhaps you or someone else can parse out sample rate capabilities from this?

**Comment by kode54** [comment]
Make a portable installation with only VGMStream, verify it's still broken. Add in your other components one by one until the problem occurs.

**Comment by rmenessec** [comment]
I just finished doing something very similar: removing components from the existing environment.

It's the crossfader DSP plugin. So far as I can tell, it's included in foo_dsp_std.dll from the core fb2k developers. Moving the location in the chain does nothing, but removing it from the active DSP plugins list immediately restores compatibility with these files. All (three) other DSP plugins I was using are compatible, or at least don't lead to fb2k locking up.

So far as I can recall, this is the first time I've ever seen the crossfader cause an incompatibility with something; it even works perfectly with NSFs inside Zip archives. (I've used fb2k on and off for years; this is the first time I went looking for 2SF and similar formats to go with your decoder plugin.)

Would getting compatibility with the crossfader plugin be an easy fix, or is this the only possible one?
